[Ready] Infection Gamemode

Whoneedspacee opened the pull request:
## About The Pull Request

**SPRITES BY THE GOD FURY MCFLURRY**

Adds infection gamemode and the infection commander antagonist. The infection commander is a unique antagonist of which there can only be one of at a time.

The infection commander spawns his core on a starting point with surrounding defenses that slightly encroaches on the stations border walls. He starts with at least 5 other players that control evolving slimes, and the gamemode can only start if there are at least 50 players on the server, as well as 6 eligible antagonists.

There are beacons that he must destroy by attempting to expand with nodes on them, and the player controlled slimes are his main army. The infection can create more slimes from the ghost pool if player controlled human bodies are dragged to the core.

The player controlled slimes do not die but instead take time to respawn if they are killed. They generate their own points as well and can use them to upgrade into different types of slimes while they are dead. They only truly die if the core of the infection is destroyed by legendary weapons.

Legendary weapons are given from doom events which are unique to this gamemode, each event has a unique item with unique effects that are the only thing able to damage the core of the infection.

Legendary weapons mostly exist as a way to prevent the gamemode from ending early, as well as to prevent something like a singularity or a bag of holding from instantly winning the mode. This was quite the issue for blob as a gamemode so this is my attempted solution for it.

Also adds multiple components for usage in the gamemode that could probably be used in many different things outside of this gamemode.

## Why It's Good For The Game

This gamemode is more of a war of attrition type gamemode with both sides gaining and losing power depending on what the players of both sides do. I'm interested in seeing the strategies that players use when they have concrete objectives to defend and attack and multiple of them, unlike nuclear operatives which is more like a wild goose chase around the station.

I don't think this gamemode is that close to being finished, and I'm well open to any changes that are proposed from testing. Even if they are drastic ones that will affect how the entire mode is played.

I'm not sure how this gamemode will play out in this type of game in the first place, and I think I need to do testing in order to move forward with development for this mode, if it's even possible to make it fun.

[Infection v10, most recent version of the gamemode that has been recorded](https://www.youtube.com/watch?v=D6zlblhr1Oc&feature=youtu.be)

## Changelog
:cl: Joe Berry
add: Infection gamemode, a new war-type gamemode that pits players against a constantly expanding and evolving enemy. Defend the beacons and destroy the core to save the station.
imageadd: Sprites by Fury McFlurry
/:cl:

Whoneedspacee committed:
ports over all the old infection stuff to this new branch

adds infection, basically a blob rip off that will have different upgrade paths

adds beacons, block infection advancement, can only be destroyed by the infection structures directly trying to consume it

adds legendary weapons, the only things that can destroy the infection core, will be dropped by bosses

icons and sounds and whatnot for everything

adds landmarks for beacon spawns and infection spawns, beacons have directions they face as well, they defend everything behind them with a wall that blocks infection

Whoneedspacee committed:
moves beacons around to promote more teamwork positions, out in the open where everyone can guard them.

beacons now spawn when the commander mob spawns in, not when the core lands.

improves cinematic feel, the biohazard containment commander now talks to you, gives cues for events that are happening, and tells you when beacons have been destroyed, as well as gives hints about how to kill the infection.

blobber- i mean infesternauts no longer rely on factories to live anymore, and just need to stay near the infection.

infesternauts are now deleted on death, will add respawning to them later.

resource infections now only add 1 point per life proc.

the infection can now evolve, get permanent improvements, 6 at the start of the game and 1 more per beacon destroyed.

Whoneedspacee committed:
removes manual expansion, clicking now tries to open a menu on structures to do certain things to them

adds base for infection structures to be upgraded

fixes up flavortext to be less wall of texty

everything is radial menu's now because they're cool

tones down infection node default expansion rate

Whoneedspacee committed:
adds radial icons for infection menus

moves upgrading stuff to the base infection file

adds upgrade types and extra descriptions for what the upgrades do for infection structures

adds base of upgrading stuff for infection mobs and removes infesternauts

adds maximum upgrade levels for evolution

fixes flavortext and makes stuff timed properly

spores get upgrade points beacons are destroyed and start off with the maximum gotten so far

Whoneedspacee committed:
why was this not versioned

Whoneedspacee committed:
naming changes to remove blob words

no more noises when beacons get hit for the commander

increases max starting points for the infection

adds upgraded shields, max upgraded shields spawn around the core now

reduces factory upgrade cost because factories suck

reduces node upgrade cost and gives them some more buffs to make expansion less of a hassle

improves resource upgrade generation

shields upgrade into reflective blobs and then stronger reflective blobs

updates help text for infection commander

changes infection light color to be cooler looking

normal infection no longer spawn upgrade menus

icon stuff added

Whoneedspacee committed:
adds slowdown when moving in infection for non blob creatures

adds turret infection, automatically fire at nearby enemies to the infection

increases base point rate for core

doubles core expansion rate

the core now automatically builds max level turrets

spores now properly recieve their upgrade points

non factory spores (sentient spores) can no longer zombify bodies

adds new icons for turret infection

Whoneedspacee committed:
overmind check for turrets so they don't fire repeatedly even when the overmind is dead

atmos pass stuff for infection structures

Whoneedspacee committed:
increases default commander upgrade points

the core now shakes the screen when it is attacked with an effective attack

the descriptions for the special infections are now more descriptive for what they are doing and what the upgrade will do

cuts down on worthless code for the evolve menu and adds a new future section for passive effects / misc stuff

the infection now takes more explosive damage dependent on the severity of the explosion

adds a new icon for the radial menus, a plus sign used for structural upgrades

Whoneedspacee committed:
reverses orientation of beacons and infection spawn on map

fixes upgrade text for everything

fixes lighting issues

infection no longer expands to space

Whoneedspacee committed:
moves infection icons to blob icons because we werent really using it anyways and they already share a bunch of shit

reworks infection upgrading alltogether to be custom per structure

fixes infection layers to not overlay a bunch of shit it shouldnt

adds ambience to infection on entry

the infection no longer absolutely destroys mechs

adds 3 turret types, resistant, infernal, and homing.

adds framework for shields to be upgradeable through non-radial-menu ways (so it doesnt take 3000 years)

removes levels and max levels for infection structures because it was stupid and boring

infection turret bullets can no longer go past beacon walls

Whoneedspacee committed:
sets up datum base for a wayyyyyyyy easier to manage upgrades system

adds can_upgrade proc as the upgrade point version of can_buy

Whoneedspacee committed:
removes show_description in all the upper level files, probably to be replaced by a standardized version in the default file.

turret upgrades now work, homing upgrades are mostly finished, might be changed a little bit in the future.

renames upgrades list to upgrade list

adds flak bullets icon

Whoneedspacee committed:
core turrets are now max upgraded resistance turrets with faster speed bullets

infection mobs now target the dead and also now zombify properly if spawned without a factory

upgrade procs for the turrets now recieve the parent structure

adds upgrades for every turret type

client spawned spores no longer zombify

upgrades now have a do_upgrade proc instead of being snowflake handled in the upgrade menu

Whoneedspacee committed:
sets up the base for sentient spore types and upgrades

adds ambiguity to upgrades datum so it can be used for multiple things

the infection can no longer use the evolve menu before it is placed (woo no bugs)

fixes text not displaying when attempting to upgrade a already max upgraded structure

adds really bad icons for the mob types

Whoneedspacee committed:
adds sentient subtype for infection spores (why didnt i do this originally this is the problem with coding at 2 am)

adds respawn points and respawn timers for sentient infection spores

tries to fix type checking for similar infection types nearby (wouldn't stop defensive turrets from being placed next to normal turrets, which could be upgraded into defensive turrets, etc...)

show_description now does something for infection mobs and structures (displays upgrade names and their descriptions)

create_spore is now called async so it doesn't cancel other calls

Whoneedspacee committed:
adds an ishelper for infection mobs

fixes communication between infection mobs

removes useless respawn point type and replaces it with the core itself

infection mobs now have a stat for their upgrade points

infection mob respawn points can now be any node and if there are no nodes, any infection alive

infection mobs can now smash walls if they're the right type

more descriptive upgrades and adds a non-functional speed boost upgrades

create_spore no longer steals one of the commanders points

Whoneedspacee committed:
spores may now refund their upgrades

spores may now upgrade while respawning and may only upgrade next to the core or nodes

adds a new proc for spores to change their type

adds upgrade complete messages

the infection now takes more damage from bombs (it really was pitiful)

Whoneedspacee committed:
adds infection bosses file for future setup

respawning spores can now alternate their set location

changes core defense setup

adds smokey particles to the core

adds armor penetration to core turrets

removes speed upgrade since new types will override the old stuff

knockback turrets no longer throw back mechs and stronger moving atoms

fixes infection structure type checking

spores no longer spawn with the incorrect amount of upgrade points

removes ambience since it didnt even work properly (stacked with area ambience) and will be replaced with just general sounds for ambience or i'll figure out how to change the area

Whoneedspacee committed:
last update for a while maybe

Whoneedspacee committed:
adds comsigs for shit i need

upgrades are now handled by component datums basically instead of just datums because im a retard and now the code is a lot more centralized and jazz and i can do a lot more stuff

the lines cut down are deceptive btw because i had to chance a lot of datum stuff to datum/component this probably halved the snowflake code shit i had here

also uh descriptions don't output upgrades you've maxed out now which is nice

basically this is for setting the framework for the mob stuff and fixes turret stuff so i don't have to repeatedly go like "ok the turret is shooting" to every single upgrade" basically for the mob stuff "ok the thing moved ok the thing hit the thing" components are great

why am i still typing i must be really bored making gamemodes is hard

Whoneedspacee committed:
moves before fire signal to base of projectiles

removes eh beacon phrases that probably will be made later to be more accurate and helpful

adds component to prevent things from crossing the beacon walls, kills them if they're living and deletes them if they're not living

adds many different spore types and upgrades to them as well as their own upgrades

sentient spores no longer take extra damage from being exposed to high temperatures (because the infection does not work like blob and its stupid)

changes layer of infection to stop covering up things

upgrades now bold their name when being displayed

adds icons for all the new types of spores

Whoneedspacee committed:
spores and such no longer die in lockers in objects or while respawning while past the beacon walls

respawn times no longer get reset when spores change type

Whoneedspacee committed:
changes how infection cores deal damage

fixes up legendary weapons code to not look like complete shit

excaliju no longer does bonus force to all structures

replaces excaliju icon with a way more badass one

renames bosses file to events file since it's going to be more than just bosses now

Whoneedspacee committed:
infection is now actually a gamemode woooooooooo

added that baybee to the config probabilities too

reworked the infection to spawn properly as an antagonist as well

also made it clear by making it a global_var and not a list that there is only supposed to be one infection alive at a time

also made it so that commanders cannot be spawned in the initialize if there's already another commander just to make sure

also sets up the base for the doom clock events for the infection legendary weapon drops and bosses and such

Whoneedspacee committed:
infection now works properly as a gamemode

you no longer die if your loc is an object

adds events for infection

adds time to next event for events subsystem formatted

no longer double adds the key to the new antagonist or deletes the previous mob they were in

infection spores now directly spawn into the commanders contents unless there's an infection core and then it spawns them directly there

adds safety checking for infection core to not spawn multiple times

the infection core on spawning now removes sentient spores from the commanders contents

fixes runtime when rotating through infection nodes while respawning

mobs now have a stat which tells them when the next infection doom event is coming

Whoneedspacee committed:
fixes upgraded sentient spores not getting selected for doom clock events or being in the mobs list

adds debug option to work with gamemode players pick

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

# Conflicts:
#	_maps/map_files/BoxStation/BoxStation.dmm

Whoneedspacee committed:
no more reports interrupting already existing reports in infection gamemode

changes placement times on the core and messages to be less overlapped

changes doom event messages so they don't overlap

adds a pre-sound effect to the doom event

removes the fact that infection cannot expand to space

changes some blob icons

Whoneedspacee committed:
removes stupid analyzer code

removes ctrl + click shields

removes old hud creation stuff

adds actions for creation and overmind abilities instead for unlockability and just more easy organization and new adding of powers

fixes round ending stuff for the gamemode

fixes roundend report weirdness

the infection no longer starts with the ability to make a turret

renames and makes core turrets more unique rather than just max upgraded resistance turrets

you can no longer place stuff too close to the core (prevents gimmicky stuff like removing core turrets to place your own turrets, etc)

stops using transfer_to in a lot of places because they transfer abilities and powers

checks if minion types exist or boss types exist before trying to create them for doom clock events (boss only, minion only events?)

infection mob upgrade menu now cuts to the case, printing out the upgrades list and opening the menu at the same time instead of having a ton of options

Whoneedspacee committed:
adds component signals for infection specific stuff

changes point rates and point caps to be less massive by just reducing everything

adds upgrades for resource and factory structures

changes the point returns of every structure

refactors upgrade system so structure upgrades derive from structure type

upgrade descriptions now print regardless of whether they've been fully bought or not

fixes an issue with there being no name or description for homing bullets upgrade

Whoneedspacee committed:
cool entering and exiting effect for infection as well as lowering your vision in the infection making combat more difficult (might consider increasing delay of fade in and fade out to make more punishing)

Whoneedspacee committed:
adds on screen hud blocking vision when you're inside the infection

adds a forced gravity that ignores space in the component

adds a variable to atoms_movable which makes it so that object can be used as a space backup to grab onto even if it is not dense or you cant pass through it (think space vines, can pass through just slowly)

the gamemode no longer checks if beacons are destroyed since the infection commander code will handle that regardless

moves beacon crossing component to actual components folder

infection structures can now have a building time so you can't just instantly box someone in with shields for example

adds actual core types so they can be handled seperately from normal types

adds a max upgrade proc for instant max upgrades

Whoneedspacee committed:
no more snowflake checks for chasms in the component, now the infection checks on being pulsed if there's a chasm and changes it to an empty space turf if it is

no more components storing their parent in a different var, also removed a lot of worthless code in the upgrades file

the core has more spread of a defense than before

comsig life for infection is now sent in the base proc instead of being spread out everywhere stupidly

infection mobs now generate their own upgrade points as well as have a hud for them instead of sharing them with the overmind

you can now override the structure build time for infection structures

infection mobs now properly update their icons on life

Whoneedspacee committed:
new epic icons & moves infection icons back to their own file

smoothing is coming for the epic icons

no more core smoke as it doesn't fit the theme of the infection anymore (crystal)

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

# Conflicts:
#	code/__DEFINES/components.dm
#	code/modules/projectiles/projectile.dm

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
icon changes

commander no longer calls life

announcement changes

upgrades system moved to straight datum and adds components

Whoneedspacee committed:
wait what are those even supposed to be there

Whoneedspacee committed:
lots of weird diffs just kind of helping my sanity to make sure im not messing something up

Whoneedspacee committed:
1 is true and 0 is false

forced_gravity component subtypes are stupid

proper loops and checking formatting in the gamemode

initialize as early as possible and destroy as late as possible

default infection color is now pure white

core now has a big 96x96 icon

no more core snowflake explosion handling

upgrade_types mostly replaced with upgrade_subtype which is much neater and less work when new upgrades are made

icon directories and files moved around again

point return for resource generators is now functional

shields now smooth properly with themselves and have much more dull colors for the crystals as well as adding a missed crystal type

Whoneedspacee committed:
typesof god im an idiot

Whoneedspacee committed:
checks movableatom for beacon crossing component

doom event mobs now die once a new event is created

no more snowflake explosions

adds more actions and makes them have cooldown timers

the infection commander is now added to the z level client mobs list and as a result can now hear sounds properly

the infection now expands more cardinally

infections that are building now don't turn into blank space while they are building

Whoneedspacee committed:
commander processing is no longer literally broken

upgraded infection mob types now retain their spent points

fixes icon for unlock turret ability

Whoneedspacee committed:
lots of filename and directory changes mimicking the recent blob refactor in order to make working with the files less of a pain

adds new icons for infection node that are animated when they actually pulse

fixes issue with radial menus not appearing on the correct position

building infections now take the pixel positions of their icon so they're on the proper position

infection turrets no longer have the wrong blob overlay

Whoneedspacee committed:
starting changes for new ui stuff

Whoneedspacee committed:
new tgui finally works for the base infection upgrades, soon moving over everything else

Whoneedspacee committed:
splits up some stuff so it's actually readable into their own sections

finishes fully converting the radial menu system to the tgui menu system

holy crap look at that line reduction maybe i'll actually be able to get things done now

Whoneedspacee committed:
adds up new icons for resource and turret infections

adds a new define for checking if something is the same type as another thing checks if the thing they're building is as well

ctrl click for the commander now places shield infections again

building structures now start at half size instead of no size

actions no longer toggle cooldown automatically

Whoneedspacee committed:
adds beamturret and effects for it

removes low quality upgrades (ones that are cheap and would be applied to every tower), switching to a system of expensive but good upgrades for more defining gameplay, and also more tower types and such

fixes the resistances, since we're going with a crystal rock theme fire shouldn't be doing more damage than brute.

resource delay now scales like normal blob so resource infection are a bit less insane

fixes type checking for infection structures

infection spore respawning is now manual and upgrading can only be done while respawning

re-adds attack animation for infection

Whoneedspacee committed:
fixes events and infection pulsing runtimes

fixes some further formatting

reduces infection default pick chance

adds icons for beamturret and sounds, as well as turret projectile icons.

Whoneedspacee committed:
fixes up some actions costs and text

fixes a bug where the game would end before the core landed

new icons + crystal coloring for different types of infection slime mobs

the infection now starts in a templated area like the meteor shuttle slamming into the station

reduces build times for lots of infection things, and also makes some require pulsing to activate their effects

Whoneedspacee committed:
basic changes to spawn template for infection

infection is now required to spawn with more enemies to pose more of a challenge off the bat

adds some more events and makes it so minions can be weighted now

fixes action icons for the infection commander

the infection commander no longer requires nodes to place things down (though they are required for activation for some)

the infection meteor now properly destroys things when it lands

infection core pulsing now more properly covers its area

infection factory now has a sprite of its own

nodes now have a longer building time to account for them being much more effective at their job

Whoneedspacee committed:
adds reflective infection to powers and gives them an icon, basically the same thing as blob reflective things

new hud icons for basically everything, replacing all the old blob ones

changing type while respawning no longer spams you with text and sounds

normal infection no longer block atmos or block projectiles

cuts down on the infection commander help wall of text

nodes now get red as they become slower

legendary weapons now have gps signals and teleportation beacons

excaliju now makes the user able to attack faster when procced

Whoneedspacee committed:
updates map to have the beacons be slightly more back

adds a few new events including syndicate and jungle mob invasion

updates text of all announcements to be more accurate and helpful

spores now recieve half of their spent upgrade points when they revert

nodes no longer slow down over time

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

# Conflicts:
#	_maps/map_files/BoxStation/BoxStation.dmm
#	code/__DEFINES/is_helpers.dm
#	tgui/assets/tgui.js

Whoneedspacee committed:
fixes some oofs

Whoneedspacee committed:
please fix

Whoneedspacee committed:
more fixing up stupid stuff

Whoneedspacee committed:
adds numerous new upgrades and abilities to everything

Whoneedspacee committed:
adds vacuum infection

map edits to boxstation and the infection template for better spawns

adds some infection defines to the antagonists defines file

slimes at roundstart now get extra points to mess around with before the game starts

the infection commander now gets permanent upgrades every time beacons are destroyed

changes a couple event mobs around

adds the coregrab ability, teleports a bunch of items over an infection turn to around the core in order to take items away or grab bodies

core autoplace is now a define

slime not spore changes

fixes a bug with spent upgrade points on sentient slimes

infection turf no longer automatically eat everything on them when pulsed

building shields now have a proper icon

miniature nodes now properly build

the core now converts bodies around it into sentient slimes

the node now has an infection base beneath it

turret projectiles now shoot faster

additional icons

the medical and freecam upgrades are no longer toggleable since it was pointless bloat on the action hud

adds back the ability to create spores with upgrade points for the commander

Whoneedspacee committed:
edits map template for infection spawn slightly to fix an issue where the core was on top of a normal infection

events no longer handrolls its sounds to all players

infection action no longer doesn't have a cost for core grab

adds barrier infection, an infection that functions almost exactly like a shield, except things can be dragged through it, but only by infectious mobs

edits winning state icon to not be a blob icon

no more floating through infection, inertia will be stopped when you cross

increases core conversion range, fixes a bug where the infection could add a body back before it was dusted, and also compenstates the infection if a spore is not created

the vacuum no longer has ear bleeding levels of volume

Whoneedspacee committed:
moves crappy events component to actual mindcontroller component

changes hud thing for infection spore

adds infection to possible role bans for admin stuff

moves isrespawning define and allows it to work with input

makes sure that you don't teleport to incorrect or qdeleted nodes with cycling respawn

can't use abilities while respawning and allows infector spores to make reflective shields

slimes now light up to prevent being stuck in the dark

the vacuum no longer sucks you in and blob acts you if you're a living creature

adds many more minor upgrades to give a sense of progression

prepares to move infectionkiller to a component system so i dont have to have this stupid file

hostile attacks can now be cancelled with a return on the signal

reflective shields no longer fail half the time

Whoneedspacee committed:
moves door to shield file

shield infection can now be mined

adds barriers to default infection spawn

adds cancel return signal to human unarmed attack

fixes up events and gives each one a unique legendary weapon

Whoneedspacee committed:
adds the beacons starts and infection starts to basically every map

core autoplace time is now actually correct

beacon explosions now hurt the infection more than they hurt the station

adds an extra info announcement to hopefully avoid confusing people about the gamemode

barriers now generate their icons properly

reduces beacon alpha so people are able to see things beneath them

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
compile tgui.js yayyyyy so fun haha

Whoneedspacee committed:
comments for everything

infection core now explodes and hurts infection everywhere around it

Whoneedspacee commented:
I'll once again state since people hate reading walls of text, but I don't think this gamemode is finished. I don't see this as a finished mode and I'm fully welcoming for even major changes, like the core being able to take damage from non-legendary things.

This is meant to be somewhat of my gift to the players of this game for all the amazing things that it's done for me, and I'll try to make it as fun for you guys as possible. Hell even if it isn't fun I've learned so many things from making this utter piece of shit that hopefully I'll be able to do more after I'm done with it for good.

wesoda25 commented:
It seems cool but gamemodes which consist of waiting for an unchanging threat have already been done and become less enjoyable the more one plays them.

actioninja commented:
le infection has arrived

Whoneedspacee commented:
It’s a 5 minute waiting time not that long, mostly just for AI/Borg stuff, basic defenses, and all access stuff. Roundtype is announced publicly 30 seconds in.

wesoda25 commented:
I really don’t wanna instantly attack it but publicly announced gamemodes are conceptually bad and go against the nature of tg (paranoia, not knowing what awaits instantly). And its not the waiting period thats bad, but the fact that you know exactly what is awaiting for you in the next round, it gets boring. Its why so many people dislike nukeops and clockcult.

Whoneedspacee commented:
Fair enough but I don’t really see how it’s much different (at least announcement wise) compared to like wizard summoning ghosts. People seem to have fun with those rounds.

But fair, I’m trying to make the gamemode more “random” with many upgrades, and maybe in the future I’ll make it so the meteor can spawn on any side of the station.

But yeah if the gamemode concept in general sucks, and it tests horribly, I’m fully open to just closing this and shoving some of my components and crap into another PR.

lolman360 commented:
it's finally here

PKPenguin321 commented on code/modules/antagonists/infection/beacon.dm:
```diff
@@ -0,0 +1,171 @@
+*/
+/obj/structure/beacon_generator/proc/destroyed_announcement(beacons_left = GLOB.infection_beacons.len)
+	sleep(80)
+	priority_announce("A beacon has been consumed by the infection, only [beacons_left] remain.","CentCom Biohazard Division", 'sound/misc/notice1.ogg')
```
Noticed this in the video, should probably read "only 1 remain***s***" when only one beacon remains

Whoneedspacee replied:
ah crap i forgot about that thanks mate

Whoneedspacee committed:
compiles now i think?

clown legendary now actually does something as well thankfully

fixes bug where legendaries weren't even fulfilling their intended purpose holy crap

fixes beacon text being off

fixes sentient slime death runtime

Whoneedspacee commented:
new video https://www.youtube.com/watch?v=D6zlblhr1Oc&feature=youtu.be showing off legendaries and looks a bit more polished

also holy crap please ignore that bug at the end god i messed up so bad

bigfatbananacyclops commented:
please testmerge this on bagil

81Denton commented:
This is very ambitious, nice! I'm curious how this plays out as a testmerge, especially for regular crew.

Akrilla commented:
Looks really great, music suits the whole theme too

nemvar commented:
Silly request that I have because of that merge conflict. Don't add any more icons to items_and_weapons.dmi. It saves you and everyone else a headache.

Whoneedspacee commented:
fair enough, there's so many defines and shit in the same place i figured why not for that, there's only like one thing in it anyways so i'll just move it out

Whoneedspacee committed:
moves legendary weapons to their own file

fixes some legendary weapons bugs

legendary weapons now spawn if there are no slimes available

legendary weapons are now undestroyable like the nuke disk

moves legendary weapons sprites to their own file

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

PKPenguin321 commented:
![e93](https://user-images.githubusercontent.com/8345184/61987706-866da580-afcd-11e9-9489-a1e15d4518bf.jpg)

Whoneedspacee commented:
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
i love tgui.js

actioninja commented:
![image](https://user-images.githubusercontent.com/5194834/61992396-aa070f00-b012-11e9-9a05-730b3321f47e.png)

Whoneedspacee committed:
comments for summoning.dm

attempts to fix problem with superpowers bad del runtime

all slimes can now pull

all slimes now automatically chat in the group chat

fixes issue with excaliju having multiple hitsounds

sentient mobs can no longer respawn if they change type and then click respawn while inside the commander

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
stored attack item is now actually stored as an item var

increases respawn time for spores default

decreases destructive spores speed

slightly alters upgrades and moves the damage upgrade to hunter

Whoneedspacee committed:
moves the superpowers stuff back to normal because it didn't fix anything

Whoneedspacee committed:
fixes the bug with the legendary weapons being qdeleted

Whoneedspacee committed:
minor map changes, infection spawn is now more inside the station so there's less threat of atmospheric changes

FuryMcFlurry commented:
OH hey, I did sprites in this one! Can't wait to play it one day.

Whoneedspacee committed:
random loot drops and death message stuff

Bawhoppen commented:
As some other people previously mentioned, this gamemode should maybe have a different name. Infection makes it sound like it's some sort of zombie mode or somthing, and is also kind of generic-sounding for something which is more unique of a concept like this. 
My suggestion is to rename it to "Abberation," which I think will make it sound much more distinct and cool.

Whoneedspacee committed:
very basic editing of the maps to add an extra gravity generator when the beacons drop, so the station doesn't lose gravity

infection spawn no longer starts with barriers

infection side now starts with 2 extra spores

standardizes method for infection spores to be created by the infection

fixes a bug where ghosts that didn't  have a mind couldn't become infection spores

mini node and reflective shield powers now have faster build times for slimes

infection slimes now always drop at least one item

all infection now atmosblocks and creates air turf when they expand as well

removes knockback by default from turrets because it was too strong

legendary weapons are now announced when they are dropped

events now only occur once

spores can now spawn again after doom event mob control

bosses can no longer cross the beacons because it makes no sense and is way too strong

drill of legends sprite update thanks to mike murdock the legend

Whoneedspacee committed:
buffs to base spore type

all infection now block atmos and also have nice air filled turf on them

buff to excaliju proc and announcement text buff

Ordonis commented:
Infection mobs were not allowing ghosts to populate them when this got run yesterday.

Whoneedspacee commented:
already fixed, will be in the next round of tests probably tommorow

Whoneedspacee committed:
clown summoning legendary now spawns sentient clowns

adds temporary visual for core grab, and only grabs mobs now

point rate is now properly increased when a beacon is destroyed

gives commander ability to direct player slimes, and they can walk there by clicking on a chat href

infector now has a moth unique visual

slimes no longer gain points while the core has not landed

the supermatter is now eaten by the infection

infection turf now properly block all atmos

new sounds for infection turrets

reduces volume on vacuum infection again jeez that is so loud

Whoneedspacee commented:
> As some other people previously mentioned, this gamemode should maybe have a different name. Infection makes it sound like it's some sort of zombie mode or somthing, and is also kind of generic-sounding for something which is more unique of a concept like this.
> My suggestion is to rename it to "Abberation," which I think will make it sound much more distinct and cool. (Doesn't "Abberation Commander" sound neat to you?)

it does but i don't really see how it fits the mode itself.

also jeez you really gonna make me rename all this shit in the code better be a damn good name haha.

TerraGS commented:
I understand you want to wait for a meta to form, but this isn't a group of guildmates prepping for a raid. It's a server of impatient players who get tired of things quickly. My suggestion is start the infection off lower on the power scale and buff aspects accordingly, that way people are more willing to give it a chance rather than writing it off as bullshit.

Whoneedspacee commented:
> I understand you want to wait for a meta to form, but this isn't a group of guildmates prepping for a raid. It's a server of impatient players who get tired of things quickly. My suggestion is start the infection off lower on the power scale and buff aspects accordingly, that way people are more willing to give it a chance rather than writing it off as bullshit.

I'm not against changing it between tests, not sure where you got that idea. It was already massively changed since the last test, and will be for this one. I put all the complaints I saw into a list and will try to address them.

carlarctg commented:
>latejoin as doctor
>infection's taken over like one half of the station
>nobody is doing anything against it
>print a saw and slowly pick at infection spreading into the chemistry maintenance
>turret shocks me for 30 damage every second i hit the infection
>sentient slime sees me, runs at me outside of the infection and instantly crits me, with each hit giving me like 10 fire stacks 

>respawn as slime, agonizingly slow going from the core to the frontline
>dragging corpses takes literally forever, not bothering with that
>die to ERT (why do we need an ert to balance the round? what's the point of researching rnd, getting mutations, or mining for ore when you can just suicide and wait a few minutes to get a free pulse rifle?)
>no idea the closest node to the frontlines i need to respawn in, XY coords don't help at all
>waddling slowly to the frontlines
>i crash, ok
>rejoin, accidentally walk into perma because the infection makes me unable to view any landmarks
>need to leave
>come back, it crashed


not gonna lie, this sucked hard

Whoneedspacee committed:
changes all the maps so that the core doesn't land directly on the station, and there are less beacons that are more straightened out

fixes nobeaconcrossing component weirdness

all slimes now have their own camera mob to respawn from instead of rotating the spawnpoints

beacons have infinity move resist and force now

the infection commander now has freecam by default

refactors slime respawning to be way less snowflaked and buggy

the infection can now automatically eat a list of important objects

the infection now tries to acid_act before it blob_acts in order to prevent things from being eaten immediately

normal infection no longer handrolls its own ridiculously low brute resist

removes unused commander verb powers

beam turret now has much less range

reflective shields are much less strong and can be mined faster now

slightly increases base turret scan range

attempts to close the ui menu before opening a new one in slimes type change

increases legendary announcement text, changes the sound, and it only occurs if the infection core is alive

doom event bosses no longer double die and drop two legendaries if they die before the next event occurs

optimumtact commented:
@tgstation/commit-access closing in 24 hours unless someone merges

SpaceManiac commented:
conflicts

Whoneedspacee committed:
adds no damage message for beacon with simplemobs

fixes up action cooldowns, removes voice and adds a tell for the flash

changes sentient mob loot drops

re-adds the overlay and fade time is halved

core landing has a new sound

shields have vastly different resistances and drop ores now when mined

igniter no longer adds a ridiculous amount of firestacks

legendary weapons are slightly stronger now

changes jungle event boss and enemies

fixes drill of legends sprite

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
fixes respawn time and being unable to see the respawn mob camera

increases acid act

shields now actually drop ores

infection events no longer affect clientless mobs to avoid clutter

Whoneedspacee commented:
well there ya go its done for now, still a few changes that i'd like to make in the future like simplified controls for the commander but the basic gameplay should be fine and at least much more tolerable now.

81Denton commented:
@optimumtact This game mode has a proper concept and looks fun. I'm sadly unable to review it in the next 2-3 weeks.

optimumtact commented:
then he can make a new pr in 2-3 weeks

Whoneedspacee commented:
fair enough, gives me more time to add the few things i wanted to add anyways

ExcessiveUseOfCobblestone commented:
@tralezab this looks similar to ur side-antag thing, perhaps u can share ideas?

Whoneedspacee committed:
fixes a few weird things i did

Whoneedspacee committed:
adds overlay for when things in the infection are being destroyed

removes weird eye mass icon for infection win

infection mobs require an upgrade to pull objects again

objects now just take damage instead of having acid acted on them or being blob acted

turrets now do much less damage (they should be moreso support towers rather than their own thing)

legendary weapons can no longer be fulled by infection faction mobs

fixes some upgrade text and infector spores can now get an upgrade to pull things again

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
reduces doom event times

fixes bug with core grab that drained points with no payout

makes the announcement messages much better

slightly reduces infection slime health

doubles respawn time for the sentient slimes, and they also now take damage when too far away from the infection

damaging effects no longer appear on temp visuals and other things for the infection

shields now drop ores when destroyed, not just when they're mined

improves clown legendary mob types and color

legendary weapons now notify ghosts when they're created

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
fixes a major bug with the tgui

infection mobs no longer have their health adjusted while respawning

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

Whoneedspacee committed:
fixes dumb tgui again

ExcessiveUseOfCobblestone reviewed (commented):
i'm like halfway lol

Whoneedspacee committed:
review fixes

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into infectionnew

increases object damage of slimes

Whoneedspacee committed:
slight map changes

Whoneedspacee commented:
:man_shrugging:

ExcessiveUseOfCobblestone commented:
ill do bottom up soon lmfao

Takov-Mala commented:
So, DGL had us test the mode out. I actually kinda liked it a bit but there's more than a few things that need fixing **big time**
The infection really doesn't need the bosses spawning, it's enough of a threat on its own. Seedlings are OP as fuck, pretty much instakills. Having to march through a region of respawning enemies full of tower defense turrets isn't exactly fun or feasible for the crew to manage. Gaining power via bodies means the infection just has to rush and kill some random fools to be large enough to have already won.

As an aside, with Cobbychem it's pretty much impossible as effectively there's very little healing that anyone is able to get once the first aid kits run out. Crew also pretty much are guaranteed to lose once a beacon is lost imo

Whoneedspacee commented:
>no bosses

i can agree with this, but i like the legendary weapons they drop and i’m not sure how else to distribute them. definitely should be focusing more on the main slime gameplay rather than custom bosses at least.

>gaining power via bodies

is yes too strong, i do like the ghost roles so i’m not 100% sure how to bring them in still, but it’s too much yeah.

>lose once a beacon is lost

yeah i wasn’t really expecting the space dragon to destroy like half the stations research but i should have expected it. i guess i just imagined people moving departments but that’s kinda lame too. doing away with bosses and increasing damage being away from the infection for slimes would help a bit.

ExcessiveUseOfCobblestone commented on code/modules/antagonists/infection/infection/structures/node.dm:
```diff
@@ -0,0 +1,88 @@
+/*
+	Changes the mini node to a normal infection after its life cycle has ended
+*/
+/obj/structure/infection/node/mini/proc/change_to_normal()
```
use addtimers/callbacks wtf

ExcessiveUseOfCobblestone commented on code/modules/antagonists/infection/infection/normalinfection.dm:
```diff
@@ -0,0 +1,495 @@
+	Life, pretty much the same as process
+*/
+/obj/structure/infection/proc/Life()
+	return
```
why are you not just using process?

ExcessiveUseOfCobblestone commented on code/modules/antagonists/infection/infection/mobs/infection_mobs.dm:
```diff
@@ -0,0 +1,164 @@
+
+/mob/living/simple_animal/hostile/infection/CanPass(atom/movable/mover, turf/target)
+	if(istype(mover, /obj/structure/infection) || istype(mover, /obj/item/projectile/bullet/infection))
+		return 1
```
mom

davethwave commented:
Just a thought on improving crewside of things. How about each time you kill a spore thing they drop a item that can permanently improve the crew member like +5 health or resistance or something like that so both sides improve with rare drops like some lavaland loot that would be useful so the commander and spores have to be a bit worried about every death as it improves the enemy side instead of rushing head on.

ShizCalev commented:
conflict

Whoneedspacee commented:
update comin out pretty soon will fix along w that

Whoneedspacee commented:
i really need a break to just not think about anything tg related and relax. ss13 in general feels like its been a drain on my life for quite some time, pretty much every part of my life brings me some joy, and i just feel like crap whenever i participate here now.

thanks to everyone who cared, maybe i’ll come back and finish this at some point.

PKPenguin321 commented:
No worries. Come back whenever you feel like it man

Akrilla commented:
mood
